src/include/ipc/probe.h

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+/**
+ * \brief Definitions of shifts and masks for format encoding in probe
+ *	  extraction stream
+ *
+ * Audio format from extraction probes is encoded as 32 bit value. Following
+ * graphic explains encoding.
+ *
+ * A|BBBB|CCCC|DDDD|EEEEE|FF|GG|H|I|J|XXXXXXX
+ * A - 1 bit - Specifies Type Encoding - 1 for Standard encoding
+ * B - 4 bits - Specify Standard Type - 0 for Audio, 1 for Video
```

Comment:
we should probably remove the video for now, since it looks like a placeholder and there's no associated format?

Reply:
obsolete documentation removed